src/main/java/com/dataguise/saas/dto/SchedulerDTO.java

```diff
@@ -0,0 +1,170 @@
+package com.dataguise.saas.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.client.model.DgTaskScheduleInfoBean;
+import io.swagger.client.model.DgTaskScheduledExecutionsBean;
+
+import java.util.List;
+
+public class SchedulerDTO {
+
+    @JsonIgnoreProperties({"connectionName", "taskId", "taskName", "taskType", "lastTaskInstanceTimeEnd",
+        "lastTaskInstanceTimeStart", "lastTaskInstanceStatus", "taskIds"})
+    private DgTaskScheduleInfoBean schedule = null;
+    private DgTaskScheduledExecutionsBean scheduleSummary = null;
+    private String sourceModule;
+    private List<TaskParamsDTO> taskParams = null;
+    private List<String> scanLocations;
+
+    private String scheduleName = null;
+
+    public void setSourceModule(String sourceModule) {
+        this.sourceModule = sourceModule;
+    }
+
+    private String scheduleType = null;
+
+    private List<String> scheduleDay = null;
+
+    //ISO DATE FORMAT
+    private String startDateTime = null;
+
+    //Weekly/Daily using hourly interval 24:00
+    private String scanType = null;
+
+    private String scheduleInterval;
+
+    //For now we fix this. No need this from UI.
+    private String scheduleScanEndDate = null;
+
+    private Boolean scheduledEventEnabled = null;
+
+    private Integer id = null;
+
+    private String moduleType = null;
+
+    @JsonIgnore
```

Comment:
**Reporter**: Checkstyle    
    **Rule**: com.puppycrawl.tools.checkstyle.checks.design.DesignForExtensionCheck
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/dto/SchedulerDTO.java L48
    
    Class &#39;SchedulerDTO&#39; looks like designed for extension (can be subclassed), but the method &#39;getTaskParams&#39; does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class &#39;SchedulerDTO&#39; final or making the method &#39;getTaskParams&#39; static/final/abstract/empty, or adding allowed annotation for the method.
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1451220198>*